src/include/packager/repositories_include.rb

```diff
@@ -469,5 +327,185 @@ def autorefresh_for?(url)
       log.info "Autorefresh flag for '#{protocol}' URL protocol: #{autorefresh}"
       autorefresh
     end
+
+  private
+
+    # scan the repository URL and return the available products
+    # @return [Array<Y2Packager::ProductLocation>] Found products
+    def scan_products(expanded_url, original_url)
+      new_repos = Pkg.RepositoryScan(expanded_url)
+      found_products = new_repos.map { |r| Y2Packager::ProductLocation.new(r[0], r[1]) }
+      log.info("Found products: #{found_products}")
+
+      # add at least one product if the scan result is empty (no product info available)
+      # to try adding the repository at the root (/) of the medium
+      if found_products.empty?
+        url_path = URL.Parse(original_url)["path"]
+        p_elems = url_path.split("/")
+
+        fallback = _("Repository")
+
+        if p_elems.size > 1
+          url_path = Ops.get(
+            p_elems,
+            Ops.subtract(Builtins.size(p_elems), 1),
+            fallback
+          )
+
+          if url_path.nil? || url_path == ""
+            url_path = Ops.get(
+              p_elems,
+              Ops.subtract(Builtins.size(p_elems), 2),
+              fallback
+            )
+
+            url_path = fallback if url_path.nil? || url_path == ""
+          end
+        elsif url_path == "/"
+          url_path = fallback
+        end
+
+        found_products << Y2Packager::ProductLocation.new(url_path, "/")
```

Comment:
own method? this is already quite long. so whole this if can be own method and it can look like
```
found_products = [single_product(original_url)] if found_products.empty?

log.info("Found products: #{found_products}")

return found_products
```